framework/src/base/MooseApp.C

```diff
@@ -138,7 +138,8 @@ MooseApp::MooseApp(InputParameters parameters) :
     _check_input(getParam<bool>("check_input")),
     _restartable_data(libMesh::n_threads()),
     _multiapp_level(0),
-    _use_legacy_output_syntax(getParam<bool>("use_legacy_output_syntax"))
+    _use_legacy_output_syntax(getParam<bool>("use_legacy_output_syntax")),
+    _cached_backup(NULL)
```

Comment:
So I did this too the first time I worked with shared pointers too. As you noticed, it compiles with C++11 but not with Boost. John did some looking and we found out that you don't have to initialize either one of these objects with NULL. It is properly initialized by its own default constructor so this statement is unnecessary.